### PR TITLE
feat(android): update gradle to 8.0.2 and gradle plugin to 8.0.0

### PR DESCRIPTION
**Pull request by IT-MikeS**
Dev build for this PR is tagged as: 

~~`@capacitor/{core,cli,android,ios}@5.0.0-dev-20230412T155745.0`~~

~~`@capacitor/{core,cli,android,ios}@5.0.0-dev-20230414T153043.0`~~

`@capacitor/{core,cli,android,ios}@5.0.0-dev-20230414T155819.0`

### CODE REVIEW

**Comment by markemer** [comment]
We'll have to make sure to update the migrate script if and when we go with this.

**Review by jcesarmobile — CHANGES_REQUESTED** [review]
you also need to update the wrapper files by running `./gradlew wrapper --distribution-type all --gradle-version 8.0.2 --warning-mode all`, but in the android-template it can by tricky since it's missing files

**Comment by markemer** [comment]
We'll also need to either update the migrate script here or in another PR.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -137,6 +137,11 @@ jobs:
       - uses: actions/setup-node@v3
         with:
           node-version: 18.x
+      - name: set up JDK 11
```
Comment (jcesarmobile) [inline .github/workflows/ci.yml:140]:
```suggestion
      - name: set up JDK 17
```

**Comment by markemer** [comment]
> Yeah, this pr is only for the template, we need another one updating capacitor android and another updating the migrate command, and the plugins too

Not to be that guy asking for homework, but should we add subtasks or a bunch of tickets for this stuff?

**Review by jcesarmobile — APPROVED** [review]

**Review by markemer — APPROVED** [review]

**Pull request merged**